### PR TITLE
Fix all integration tests [TESTING]

### PR DESCRIPTION
**Pull request by fassadlr**
Just testing to see how many still fail...

### CODE REVIEW

File: src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs [hunk 1]
```diff
@@ -78,9 +78,9 @@ public void Dispose()
         /// Executes start or stop method of all the features registered with the associated full node.
         /// </summary>
         /// <param name="callback">Delegate to run start or stop method of the feature.</param>
-        /// <param name="reverseOrder">Reverse the order of which the features are executed.</param>
+        /// <param name="disposing">Reverse the order of which the features are executed.</param>
```
Comment (bokobza) [inline src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs]: Why the change of name?

File: src/Stratis.Bitcoin.Tests.Common/TestBase.cs [hunk 2]
```diff
@@ -86,7 +86,7 @@ public static string GetTestDirectoryPath(object caller, [System.Runtime.Compile
         /// <returns>The path of the directory.</returns>
         public static string GetTestDirectoryPath(string testDirectory)
         {
-            return Path.Combine("TestCase", testDirectory);
+            return Path.Combine(@"..\..\..\..\TestCase", testDirectory);
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Tests.Common/TestBase.cs]: Where is that leading to? `src`? Can you explain why is this good?

Comment (bokobza) [reply src/Stratis.Bitcoin.Tests.Common/TestBase.cs]: The Mac and Linux builds have trouble with this path I think.
Maybe try 
```suggestion
            return Path.Combine("..", "..", "..", "..", "TestCase", testDirectory);
```

Comment (fassadlr) [reply src/Stratis.Bitcoin.Tests.Common/TestBase.cs]: Fixed

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerFailedReorgTests.cs [hunk 3]
```diff
@@ -230,9 +246,11 @@ public void ReorgChain_FailsFullValidation_ChainHasBlocksAndHeadersOnly_NodesDis
         {
             using (var builder = NodeBuilder.Create(this))
             {
-                var minerA = builder.CreateStratisPosNode(new StratisRegTest()).NotInIBD().WithWallet().Start();
-                var minerB = builder.CreateStratisPosNode(new StratisRegTest()).NotInIBD().WithWallet().Start();
-                var minerC = builder.CreateStratisPosNode(new StratisRegTest()).NotInIBD().NoValidation().WithWallet().Start();
+                var stratisNoValidationRulesNetwork = new StratisNoValidationRulesNetwork();
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerFailedReorgTests.cs]: Fix here!

File: src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs [hunk 4]
```diff
@@ -120,13 +120,12 @@ public void MempoolReceiveFromManyNodes()
             }
         }
 
-        [Retry(2)]
-        [Trait("Unstable", "True")]
+        [Fact(Skip = "Unstable")]
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs]: Change back to unstable:

File: src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMiningTests.cs [hunk 5]
```diff
@@ -8,14 +9,22 @@ namespace Stratis.Bitcoin.IntegrationTests.Miners
 {
     public class ProofOfStakeMiningTests
     {
+        private class StratisRegTestLastPowBlock : StratisRegTest
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMiningTests.cs]: Maybe move these to a base class

File: src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs [hunk 6]
```diff
@@ -159,15 +149,16 @@ public void CanCoreSyncFromStratis()
             }
         }
 
-        [Retry(2)]
+        [Fact]
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs]: Change back to Retry

File: src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs [hunk 7]
```diff
@@ -242,16 +233,15 @@ public void PullerVsMinerRaceCondition()
                 TestHelper.Disconnect(stratisMinerLocal, stratisMinerRemote);
 
                 // Mine block C2p.
-                TestHelper.MineBlocks(stratisMinerRemote, 1);
+                TestHelper.MineBlocks(stratisMinerRemote, 1, true);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs]: Confirm if true or false

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingToAndFromManyAddressesSpecification_Steps.cs [hunk 8]
```diff
@@ -17,100 +17,111 @@ namespace Stratis.Bitcoin.IntegrationTests.Wallet
...
         private const string WalletName = "mywallet";
         private const string WalletPassword = "password";
-        private const string WalletPassphrase = "passphrase";
         private const string WalletAccountName = "account 0";
         private const int UnspentTransactionOutputs = 50;
 
         public SendingToAndFromManyAddressesSpecification(ITestOutputHelper outputHelper) : base(outputHelper) { }
 
-        protected override void BeforeTest()
+        private class RegTestOverrideCoinbaseMaturity : BitcoinRegTest
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingToAndFromManyAddressesSpecification_Steps.cs]: Move to helper class

File: src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletAddressGenerationAndFundsVisibility_Steps.cs [hunk 9]
```diff
@@ -51,14 +58,11 @@ public Wallet_address_generation_and_funds_visibility(ITestOutputHelper output)
         private void MineSpendableCoins()
         {
             this.sendingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity.Should().Be(this.receivingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity);
-            this.previousCoinBaseMaturity = this.sendingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity;
-
-            var coinbaseMaturity = 1;
 
-            this.sendingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity = coinbaseMaturity;
-            this.receivingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity = coinbaseMaturity;
+            this.sendingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity = 1;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletAddressGenerationAndFundsVisibility_Steps.cs]: This isnt needed any more

**Comment by fassadlr** [comment]
@bokobza latest build passed:

https://dev.azure.com/StratisProject/StratisBitcoinFullNode/_build/results?buildId=1485&view=results

Merge when ready 👍

**Comment by bokobza** [comment]
Thanks @fassadlr looking now.

File: src/Stratis.Bitcoin.Features.Consensus.Tests/ProvenBlockHeaders/ProvenBlockHeaderStoreTests.cs [hunk 10]
```diff
@@ -99,7 +99,7 @@ public async Task AddToPending_Adds_To_CacheAsync()
             outHeaderRepo.FirstOrDefault().Should().BeNull();
         }
 
-        [Fact]
+        [Fact(Skip = "Unstable")]
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Consensus.Tests/ProvenBlockHeaders/ProvenBlockHeaderStoreTests.cs]:
```suggestion
        [Fact]
        [Trait("Unstable", "True")]
```

File: src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs (hunk 4 above)
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs]:
```suggestion
        [Fact]
        [Trait("Unstable", "True")]
```

File: src/Stratis.Bitcoin.Tests/Utilities/PeriodicTaskTest.cs [hunk 11]
```diff
@@ -13,15 +13,15 @@ public class PeriodicTaskTest : LogsTestBase
...
         /// Unable to tests background thread exception throwing due to https://github.com/xunit/xunit/issues/157.
-        /// </remarks>
+        /// </summary>
         public PeriodicTaskTest() : base()
         {
             this.iterationCount = 0;
         }
 
-        [Fact]
+        [Fact(Skip = "Unstable")]
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Tests/Utilities/PeriodicTaskTest.cs]:
```suggestion
        [Fact]
        [Trait("Unstable", "True")]
```

**Comment by bokobza** [comment]
@fassadlr I changed some of the skipped tests to be unstable instead.
Also, I don't see why we need to have all these new Networks classes if all they do is give the network a different name. 
In the ones that only change the name, surely if we do new StratisregTest() we shouldn't have any trouble?
Maybe in a next PR we can review that.

**Comment by bokobza** [comment]
image
